### ISSUE TITLE
Panic:runtime error: index out of range after upgrading to 1.3 to 1.6

### ISSUE THREAD

**Issue by hardiksondagar** [influxdata/influxdb#10128]
__System info:__ [1.6, AWS EC2 m4.xlarge 16GB RAM Ubuntu 16.04, 256GB SSD gp2 3000 IOPS ]

__Steps to reproduce:__

1. Querying one series data
```
> SELECT "SensorId","2002" FROM Reading WHERE "SensorId"='17001' AND time > now()-1m ORDER BY time DESC tz('Asia/Kolkata');
name: Reading
time                      SensorId 2002
----                      -------- ----
2018-07-25T21:05:53+05:30 17001    1.6144
2018-07-25T21:05:38+05:30 17001    1.5099
2018-07-25T21:05:23+05:30 17001    1.7591
2018-07-25T21:05:08+05:30 17001    2.2847
```
2. Querying 100 serieses data
```
> SELECT "SensorId","2002" FROM Reading WHERE "SensorId" =~ /^1700[0-9]{1}$/ AND time > now()-1m ORDER BY time DESC LIMIT 5 tz('Asia/Kolkata');
name: Reading
time                      SensorId 2002
----                      -------- ----
2018-07-25T21:03:11+05:30 17009    3.0074
2018-07-25T21:03:11+05:30 17008    47.5647
2018-07-25T21:03:10+05:30 17006    38.8565
2018-07-25T21:03:10+05:30 17005    25.7809
2018-07-25T21:03:10+05:30 17007    22.3157
```
3. Querying 1000 serieses data
```
> SELECT "SensorId","2002" FROM Reading WHERE "SensorId" =~ /^17[0-9]{3}$/ AND time > now()-1m ORDER BY time DESC LIMIT 5 tz('Asia/Kolkata');

ERR: SELECT SensorId, "2002" FROM smartsense.autogen.Reading WHERE SensorId =~ /^17[0-9]{3}$/ AND time > now() - 1m ORDER BY time DESC LIMIT 5 TZ('Asia/Kolkata') [panic:runtime error: index out of range]
```

__Expected behavior:__ It should return data

__Actual behavior:__ Throwing error `panic:runtime error: index out of range`, the same query was giving result in 1.3 version

__Additional info:__ `SensorId` is tag and `2002` is field

__System info:__
[iostat.txt](https://github.com/influxdata/influxdb/files/2231100/iostat.txt)
[vars.txt](https://github.com/influxdata/influxdb/files/2231101/vars.txt)
[profiles.tar.gz](https://github.com/influxdata/influxdb/files/2231102/profiles.tar.gz)

**Comment by e-dard** [comment]
@hardiksondagar Hi, this is a duplicate of https://github.com/influxdata/influxdb/issues/10069. It has been fixed in `master`, and is also under the `1.6.1rc1` tag. 

The fix will be available in the `1.6.1` release, which isn't quite ready yet.

**Comment by e-dard** [comment]
@hardiksondagar FYI, it's because of your repeat operator `{3}` in your regex. If you can rewrite the regex to only have two or fewer repeats `{2}`, or you can rewrite it a different way to not use the repeat operator, then you will not hit the panic and can carry on using 1.6.0 successfully until `1.6.1` is released 😃 

Hope that helps.
